radiation/python/radsw/setcoef_gt4py.py

```diff
@@ -0,0 +1,271 @@
+import serialbox as ser
+
+ddir = "/Users/AndrewP/Documents/work/physics_standalone/radiation/fortran/radsw/dump"
+serializer = ser.Serializer(ser.OpenModeKind.Read, ddir, "Serialized_rank1")
+savepoints = serializer.savepoint_list()
+
+rebuild = False
+validate = True
+backend = "gtc:gt:cpu_ifirst"
+
+invars = ["pavel", "tavel", "h2ovmr", "nlay", "nlp1"]
+
+indict = dict()
+for var in invars:
+    tmp = serializer.read(var, serializer.savepoint["swrad-setcoef-input-000000"])
+    if var in ["pavel", "tavel", "h2ovmr"]:
+        indict[var] = np.tile(tmp[None, None, :], (npts, 1, 1))
+    else:
+        indict[var] = tmp[0]
+
+indict_gt4py = dict()
+for var in invars:
+    if var in ["pavel", "tavel", "h2ovmr"]:
+        indict_gt4py[var] = create_storage_from_array(
+            indict[var], backend, shape_nlay, DTYPE_FLT
+        )
+    else:
+        indict_gt4py[var] = indict[var]
+
+outvars_flt = [
+    "fac00",
+    "fac01",
+    "fac10",
+    "fac11",
+    "selffac",
+    "selffrac",
+    "forfac",
+    "forfrac",
+]
+
+outvars_int = ["indself", "indfor", "jp", "jt", "jt1"]
+
+outdict_gt4py = dict()
+
+for var in outvars_flt:
+    outdict_gt4py[var] = create_storage_zeros(backend, shape_nlay, DTYPE_FLT)
+for var in outvars_int:
+    outdict_gt4py[var] = create_storage_zeros(backend, shape_nlay, DTYPE_INT)
+
+outdict_gt4py["laytrop"] = create_storage_zeros(backend, shape_nlay, DTYPE_BOOL)
+
+locvars = ["plog", "fp", "fp1", "ft", "ft1", "tem1", "tem2", "jp1"]
+
+locdict_gt4py = dict()
+for var in locvars:
+    if var == "jp1":
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlay, DTYPE_INT)
+    else:
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlay, DTYPE_FLT)
+
+ds = xr.open_dataset("../lookupdata/radsw_ref_data.nc")
+preflog = ds["preflog"].data
+preflog = np.tile(preflog[None, None, None, :], (npts, 1, nlay, 1))
+tref = ds["tref"].data
+tref = np.tile(tref[None, None, None, :], (npts, 1, nlay, 1))
+lookupdict_gt4py = dict()
+
+lookupdict_gt4py["preflog"] = create_storage_from_array(
+    preflog, backend, shape_nlay, (DTYPE_FLT, (59,))
+)
+lookupdict_gt4py["tref"] = create_storage_from_array(
+    tref, backend, shape_nlay, (DTYPE_FLT, (59,))
+)
+
+stpfac = 296.0 / 1013.0
+
+
+@stencil(backend=backend, rebuild=rebuild, externals={"stpfac": stpfac})
+def setcoef(
+    pavel: FIELD_FLT,
+    tavel: FIELD_FLT,
+    h2ovmr: FIELD_FLT,
+    laytrop: FIELD_BOOL,
+    jp: FIELD_INT,
+    jt: FIELD_INT,
+    jt1: FIELD_INT,
+    fac00: FIELD_FLT,
+    fac01: FIELD_FLT,
+    fac10: FIELD_FLT,
+    fac11: FIELD_FLT,
+    selffac: FIELD_FLT,
+    selffrac: FIELD_FLT,
+    indself: FIELD_INT,
+    forfac: FIELD_FLT,
+    forfrac: FIELD_FLT,
+    indfor: FIELD_INT,
+    plog: FIELD_FLT,
+    fp: FIELD_FLT,
+    fp1: FIELD_FLT,
+    ft: FIELD_FLT,
+    ft1: FIELD_FLT,
+    tem1: FIELD_FLT,
+    tem2: FIELD_FLT,
+    jp1: FIELD_INT,
+    preflog: Field[(DTYPE_FLT, (59,))],
+    tref: Field[(DTYPE_FLT, (59,))],
```

Comment:
what is '59', can we use a variable for this?

Reply:
Unfortunately not, they are hardcoded in the Fortran too